server/app/views/components/TextFormatter.java

```diff
@@ -98,16 +98,51 @@ private static String addTextSize(String markdownText) {
   }
 
   private static String addRequiredIndicator(String markdownText) {
+    // If the question ends with a list (UL or OL tag), we need to handle the required indicator
+    // differently
+    if (endsWithListTag(markdownText, "</ul>\n")) {
```

Comment:
question (non-blocking):  Would it be possible for the markdown text to end with a closing list tag but no new line?  If it's not possible, can you help me understand why there's always a new line at the end?

Reply:
Both markdown parsers (the java one and the javascript one) add new line characters at the end of block elements (paragraphs, lists, etc.). Maybe all markdown parsers do this? I'm guessing this is so the next block element starts below the last one. New line characters aren't inserted after inline elements (em, strong, span), but those are always within block elements.

Reply:
Got it.  Thanks for explaining.